main/src/ca/uwaterloo/flix/language/phase/Kinder.scala

```diff
@@ -114,35 +114,48 @@ object Kinder {
    */
   private def visitStruct(struct0: ResolvedAst.Declaration.Struct, taenv: Map[Symbol.TypeAliasSym, KindedAst.TypeAlias], root: ResolvedAst.Root)(implicit flix: Flix): Validation[KindedAst.Struct, KindError] = struct0 match {
     case ResolvedAst.Declaration.Struct(doc, ann, mod, sym, tparams0, fields0, loc) =>
-      // if not annotated tparams are assumed to be Star except the last one which is Eff
-      val tparams1 = tparams0 match {
+      // In the case in which the user doesn't supply any type params,
+      // the parser will have already notified the user of this error
+      // The recovery step here is to simply add a single type param that is never used
+      val tparams1 = if (tparams0.tparams.isEmpty) {
+        val regionTparam = ResolvedAst.TypeParam.Unkinded(Name.Ident("", loc), Symbol.freshUnkindedTypeVarSym(Ast.VarText.Absent, true, loc), loc)
```

Comment:
We probably want to give it some name, not just the empty string, at least for debugging purposes.

How about `_rc`.

Reply:
Counterintuitively, the `isRegion` parameter here should actually be `false`. It's because we're not defining a literal region here, but rather a region variable.

Reply:
Btw we should almost always use named parameters for Booleans (`isRegion = false`)

Reply:
What if `_rc` shadows an actual name?

Reply:
Maybe `$rc`? since variables can't have `$`